src/cpp/ext/otel/otel_plugin.h

```diff
@@ -217,53 +186,237 @@ class OpenTelemetryPluginBuilderImpl {
+          return false;
+        }
       }
+      return true;
     }
-    return true;
-  }
 
- private:
-  explicit ActivePluginOptionsView(
-      absl::FunctionRef<bool(const InternalOpenTelemetryPluginOption&)> func) {
-    for (size_t i = 0; i < OpenTelemetryPluginState().plugin_options.size();
-         ++i) {
-      const auto& plugin_option = OpenTelemetryPluginState().plugin_options[i];
-      if (plugin_option != nullptr && func(*plugin_option)) {
-        active_mask_.set(i);
+   private:
+    explicit ActivePluginOptionsView(
+        absl::FunctionRef<bool(const InternalOpenTelemetryPluginOption&)> func,
+        const OpenTelemetryPlugin* otel_plugin) {
+      for (size_t i = 0; i < otel_plugin->plugin_options().size(); ++i) {
+        const auto& plugin_option = otel_plugin->plugin_options()[i];
+        if (plugin_option != nullptr && func(*plugin_option)) {
+          active_mask_.set(i);
+        }
       }
     }
+
+    std::bitset<64> active_mask_;
+  };
+
+  class ClientScopeConfig : public grpc_core::StatsPlugin::ScopeConfig {
+   public:
+    ClientScopeConfig(const OpenTelemetryPlugin* otel_plugin,
+                      const ChannelScope& scope)
+        : active_plugin_options_view_(ActivePluginOptionsView::MakeForClient(
+              scope.target(), otel_plugin)),
+          filtered_target_(
+              // Use the original target string only if a filter on the
+              // attribute is not registered or if the filter returns true,
+              // otherwise use "other".
+              otel_plugin->target_attribute_filter() == nullptr ||
+                      otel_plugin->target_attribute_filter()(scope.target())
+                  ? scope.target()
+                  : "other") {}
+
+    const ActivePluginOptionsView& active_plugin_options_view() const {
+      return active_plugin_options_view_;
+    }
+
+    absl::string_view filtered_target() const { return filtered_target_; }
+
+   private:
+    ActivePluginOptionsView active_plugin_options_view_;
+    std::string filtered_target_;
+  };
+  class ServerScopeConfig : public grpc_core::StatsPlugin::ScopeConfig {
+   public:
+    ServerScopeConfig(const OpenTelemetryPlugin* otel_plugin,
+                      const grpc_core::ChannelArgs& args)
+        : active_plugin_options_view_(
+              ActivePluginOptionsView::MakeForServer(args, otel_plugin)),
+          injected_labels_from_plugin_options_(
+              otel_plugin->plugin_options().size()) {}
+
+    const ActivePluginOptionsView& active_plugin_options_view() const {
+      return active_plugin_options_view_;
+    }
+
+    std::vector<std::unique_ptr<LabelsIterable>>&
+    injected_labels_from_plugin_options() {
+      return injected_labels_from_plugin_options_;
+    }
+
+   private:
+    ActivePluginOptionsView active_plugin_options_view_;
+    std::vector<std::unique_ptr<LabelsIterable>>
+        injected_labels_from_plugin_options_;
+  };
+
+  struct Client {
```

Comment:
now that this struct is outside of `OpenTelemetryPluginState`, let's call this `ClientMetrics` and `ServerMetrics` respectively

Reply:
Done.